dashboard/app/models/pd/workshop_daily_survey.rb

```diff
@@ -94,7 +101,7 @@ def self.create_placeholder!(user_id:, pd_workshop_id:, day:, form_id:, submissi
     def duplicate?
       # See if this user already has a submission for this workshop, day, & form.
       # Note: this duplicate record would fail the uniqueness validation
-      self.class.exists?(attributes.slice(:user_id, :pd_workshop_id, :day, :form_id))
+      new_record? && self.class.exists?(slice(:user_id, :pd_workshop_id, :day, :form_id))
```

Comment:
Just talking this through - if we somehow created a duplicate record and committed it ignoring checks, this would be inaccurate. But that should never happen so this is okay.

Reply:
The [uniqueness validation](https://github.com/code-dot-org/code-dot-org/blob/ad146d5e47cfaf91b1725bb669fc082ddc2a4bb4/dashboard/app/models/pd/workshop_daily_survey.rb#L54-L55) would fail, so we wouldn't be able to save it anyway. This `duplicate?` check is just to detect those and skip them to prevent causing errors.

Reply:
And we can't even commit, skipping validations, because the [Sql unique constraint](https://github.com/code-dot-org/code-dot-org/blob/fe770928839ab17eb2d0a90bf7f080ce9251aab3/dashboard/db/schema.rb#L833) would block it.